Python/sysmodule.c

```diff
@@ -2923,7 +2923,11 @@ _PySys_InitMain(PyThreadState *tstate)
     SET_SYS_FROM_WSTR("exec_prefix", config->exec_prefix);
     SET_SYS_FROM_WSTR("base_exec_prefix", config->base_exec_prefix);
     {
-        PyObject *str = PyUnicode_FromString(PLATLIBDIR);
+        const char* platlibdir = getenv("_PYTHON_PLATLIBDIR");
+        if(!platlibdir) {
+            platlibdir = PLATLIBDIR;
+        }
+        PyObject *str = PyUnicode_FromString(platlibdir);
```

Comment:
It's wrong to decode an environment variable from UTF-8. You must use PyUnicode_DecodeFSDefault() for the env var, but keep PyUnicode_FromString for PLATLIBDIR.